Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp

```diff
@@ -114,12 +114,12 @@ void ProcessLauncher::launchProcess()
         argv[i++] = nullptr;
         WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
 
-        m_processID = ProcessProviderLibWPE::singleton().launchProcess(m_launchOptions, argv, WTFMove(webkitSocketPair.client));
+        m_processID = ProcessProviderLibWPE::singleton().launchProcess(m_launchOptions, argv, webkitSocketPair.client.release());
```

Comment:
Hm, I see the original code doesn't use `webkitSocketPair.client` past this point, so clearly it was expecting to transfer ownership, and here you preserve the previous behavior. But `ProcessProviderLibWPE::launchProcess` (which is hidden, it's actually in ProcessProviderGLib.cpp) does not expect to receive ownership, so we actually leak the file descriptor. That's a pretty serious preexisting bug. I also checked the PlayStation interface and it also does not receive ownership, but it closes the client fd so doesn't have this bug.

 Might as well fix it now:

```suggestion
        m_processID = ProcessProviderLibWPE::singleton().launchProcess(m_launchOptions, argv, webkitSocketPair.client.value());
```

This is important and arguably we shouldn't sneak it into a build fix commit, so you could land it separately if you prefer, or edit the commit message to mention that you're fixing more than just the build regression, whichever way seems nicer to you.

Reply:
>  But ProcessProviderLibWPE::launchProcess (which is hidden, it's actually in ProcessProviderGLib.cpp) does not expect to receive ownership, so we actually leak the file descriptor.

This is somewhat of an understatement; it just completely ignores the client file descriptor. I do not understand how the launched process is able to function without its IPC connection to the parent process. That's _really_ weird. I wonder what I'm missing?

Reply:
I applied the suggested change and updated commit message.